### PR TITLE
Yet Another Redis Cache

### PR DESCRIPTION
**Pull request by aliakb**
https://jira.biaprotect.com/jira/browse/TD-10983

### CODE REVIEW

File: yarc.gemspec [hunk 1]
```diff
@@ -20,4 +20,8 @@ Gem::Specification.new do |spec|
 
   spec.add_development_dependency "bundler", "~> 1.7"
   spec.add_development_dependency "rake", "~> 10.0"
+  spec.add_development_dependency "redis", "~> 3.0"
+  spec.add_development_dependency "multi_json", "~> 1.0"
```
Comment (jonp) [inline yarc.gemspec]: The previous two lines should probably be non-development

File: yarc.gemspec [hunk 2]
```diff
@@ -8,8 +8,8 @@ Gem::Specification.new do |spec|
   spec.version       = Yarc::VERSION
   spec.authors       = ["aliakb"]
   spec.email         = ["abaturytski@gmail.com"]
-  spec.summary       = %q{TODO: Write a short summary. Required.}
-  spec.description   = %q{TODO: Write a longer description. Optional.}
+  spec.summary       = "Yet Another Redis Cache"
+  spec.description   = "Redis-backed cache with temporary and permanent storage"
   spec.homepage      = ""
   spec.license       = "MIT"
```
Comment (jonp) [inline yarc.gemspec]: license should be "None"

File: yarc.gemspec [hunk 3]
```diff
@@ -8,8 +8,8 @@ Gem::Specification.new do |spec|
   spec.version       = Yarc::VERSION
   spec.authors       = ["aliakb"]
   spec.email         = ["abaturytski@gmail.com"]
```
Comment (jonp) [inline yarc.gemspec]: authors and email should be BIA and dev@biaprotect.com, respectively.

**Comment by jonp** [comment]
If I'm reading this correctly, the client code is responsible for migrating entries from temporary to permanent. Is that correct?

Is this going to become the replacement for pocoa's StorageManager?

File: yarc.gemspec (hunk 1 above)
Comment (aliakb) [inline yarc.gemspec]: Fixed.

File: yarc.gemspec (hunk 2 above)
Comment (aliakb) [inline yarc.gemspec]: Fixed

File: yarc.gemspec (hunk 3 above)
Comment (aliakb) [inline yarc.gemspec]: Fixed

**Comment by aliakb** [comment]
@jonp You're right. This component will be used as a cache in the storage manager. Adding a file will put it into a permanent cache; upload will take it from the permanent cache, upload it, and migrate into the temporary cache. The cache object will have a method for migrating the key from a permanent storage to the temporary one (maybe in the opposite direction, too, if I find a scenario for that). The migration would require two steps:
1. Releasing data in the source store (removing all store-specific metadata associated with the key)
2. Migrating the key to the destination store (renaming the key and assigning it store-specific metadata, such as TTL).

Also, having two storages in the storage manager enables easy and simple implementation of "mark as processed/no longer needed" scenario. Whenever the file has been processed, we will simply remove it form the temporary storage. And the file appears in the temporary storage only after the file has been uploaded. No more flags to track, no more transactions/locking.

**Pull request merged**